### PR TITLE
feat(quarantine): park orphaned nodes for review instead of losing them

### LINKED ISSUE #266: Feature: quarantine orphaned nodes for periodic review instead of deleting
## Context

Measured on a real data home (2917 nodes / 4957 edges):

| Metric | Count | % |
|---|---|---|
| Degree-0 (orphaned) nodes | 151 | 5.2% |
| Degree-1 nodes | 1105 | 37.9% |

Orphan types: service 43, document 32, project 26, preference 13, domain 12, goal 9, value 7, person 3, skill 3, decision 2, role 1.

## Problem

There is no safe path to remove orphaned nodes today:

1. **No delete anywhere** — no MCP tool, no CLI command (`prune`/`gc` do not exist).
2. **Hand-editing `graph/facts.jsonl` is forbidden** (AGENTS.md rule #1) — it is compiled output; the next `lorekeep compile` re-extracts from `raw/` and the nodes come back.
3. **Deleting the raw source is blunt** — an orphan may be extraction noise *or* a genuinely useful fact that simply never got linked (false orphan). Destroying the raw source loses the signal needed to tell them apart later.

So orphans accumulate, and each compile keeps promoting more noise into nodes.

## Proposal: quarantine, not delete

Park orphans in a review bucket instead of deleting them, so a human can periodically re-evaluate whether each one is *truly* orphaned.

Suggested mechanics:

1. **Detection** — `lorekeep quarantine` (CLI) and/or an MCP tool that finds degree-0 nodes (configurable threshold: degree-0 by default, degree ≤ 1 opt-in given the 38% degree-1 noise mass).
2. **Marking, not mutation** — quarantining a node sets a flag/`valid_to` (or moves its id into the existing `manifest.quarantine` list, which is already present but always empty). The node and its provenance stay intact in `facts.jsonl`.
3. **Wiki exclusion** — quarantined nodes are excluded from normal wiki output, or rendered under a dedicated `quarantine/` section so they stop polluting the main graph view in Obsidian.
4. **Review workflow** — `lorekeep quarantine review` lists quarantined nodes together with the `raw/` sources that produced them, with three outcomes per node:
   - **restore** — it was a false orphan; re-activate (and optionally suggest edges).
   - **confirm delete** — remove the underlying raw source so the next compile drops it for real.
   - **keep quarantined** — leave parked.
5. **Compile persistence** — quarantine decisions must survive `lorekeep compile` (same guarantee as `props.merged_ids` today); otherwise the flag is silently reset on every recompile.
6. Optional: **TTL** — auto-purge nodes quarantined longer than N days/compiles without review, with a warning before purge.

## Design note: why a quarantine flag, not merge-into-one-node

An earlier idea was to `merge_entities` all orphans into a single "orphan review" node. That does not work:

- Cross-type merges are blocked, and orphans span 11 node types — it would need at least 11 bucket nodes.
- Merging folds 151 distinct facts (each with its own `src` provenance) into one mega-node with an unreadable description — granularity and per-fact review are lost, and un-merging is not supported.

A per-node quarantine flag keeps every fact individually inspectable and reversible.

## Open questions

- Threshold: degree-0 only, or configurable degree ≤ N?
- Should quarantine be namespace-scoped (respect the permission boundary)?
- TTL default (off / 90 days / N compiles)?
- Does `lorekeep doctor` need to validate quarantine state (e.g. flag nodes that gained an edge while quarantined)?

### PR DESCRIPTION
**Pull request by Aohkne**
## Summary

Closes #266. Adds `lorekeep quarantine detect [--apply]` and `lorekeep quarantine review` so degree-0 nodes can be parked for human review instead of resurfacing as lint noise every compile or being deleted blindly.

- Flag lives in `props` (`quarantined_at`/`quarantined_reason`), no schema migration needed.
- Survives recompiles the same way `props.merged_ids` does (`_load_prev_quarantine` + `_apply_prev_quarantine`).
- `agent lint`/`self_heal` stop reporting already-quarantined nodes.
- Wiki excludes quarantined nodes, but a node reappears automatically if it later gains an edge (re-checked at wiki-gen time, not just the persisted flag).
- Scope intentionally excludes auto-deleting the raw source on "confirm delete" — that part of #266 doesn't have a settled design yet.

## Test plan

- [x] `uv run pytest tests/test_quarantine.py` (30 new tests)
- [x] Full suite: `uv run pytest` — 1674 passed, 2 skipped (2 pre-existing, unrelated `timeout`-binary failures on this machine, reproduced on unmodified `main` too)
- [x] `test_core_regression.py`, `test_determinism.py`, `test_docs_contract.py` all green
- [x] `scripts/generate_cli_reference.py` regenerated `docs/reference/cli.md`

### CODE REVIEW

File: src/lorekeep/agent.py [hunk 1]
```diff
@@ -45,8 +45,13 @@ def issue_count(self) -> int:
 def lint(store: GraphStore) -> LintReport:
     report = LintReport()
 
-    # Orphans: nodes with zero inbound or outbound edges
+    # Orphans: nodes with zero inbound or outbound edges. Already-quarantined
+    # nodes are excluded — a human has parked them for review (see
+    # `lorekeep quarantine`), so they should stop resurfacing as lint noise.
     for nid in store.node_ids():
+        node = store.get_node(nid)
```
Comment (manhhailua) [inline src/lorekeep/agent.py]: `store.node_ids()` includes NetworkX phantom endpoints from dangling edges. Those nodes have no `"node"` attr, so `get_node` raises `KeyError: 'node'` before the orphan check runs.

`quarantine detect` calls `lint().orphans`, so the same crash hits this feature's CLI.

Iterate `store.all_nodes()` (phantoms already filtered; this is what `self_heal` uses) and skip `is_quarantined(n)`. Add a test with `GraphStore([svc:a], [a→ghost])` that `lint` returns instead of raising.

**Review by manhhailua — CHANGES_REQUESTED** [review]
The direction matches #266: park degree-0 nodes with a props flag (not a schema bump, not a mega-merge), persist it across `compile` the same way `merged_ids` is persisted, hide them from wiki while they stay degree-0, and leave `raw/` / confirm-delete for a later design. Tests cover detect/review, lint/heal exclusion, wiki reappearance after an edge appears, and a real `compile` round-trip. Docs + generated CLI reference are in the same PR. Merge onto current `main` is clean; required CI is green.

One merge-blocking crash, plus a couple of follow-ups that should land with the fix.

## Must fix

`lint()` now calls `store.get_node(nid)` while iterating `store.node_ids()`. `node_ids()` includes NetworkX phantom endpoints created by dangling edges; those nodes have empty node data, so `get_node` raises `KeyError: 'node'`. I reproduced this on the PR tip:

```text
GraphStore([svc:a], [edge a → svc:ghost])
lint(store)  →  KeyError: 'node'
```

That is a regression: `agent lint` / `quarantine detect` (which reuses `lint().orphans`) now crash on the exact broken graphs lint/heal exist to diagnose. Compile output usually has no dangling edges (resolve drops them), but `self_heal` and `lint --auto-fix` exist because they still show up from journals / partial writes.

Fix: iterate `store.all_nodes()` (already filters phantoms; `self_heal` uses the same set) instead of `node_ids()` + `get_node`:

```python
for n in store.all_nodes():
    if is_quarantined(n):
        continue
    if not store.out_edges(n.id) and not store.in_edges(n.id):
        report.orphans.append(n.id)
```

Add a regression test: `GraphStore([svc:a], [edge a→ghost])` — `lint` must return, not raise, and must not treat `svc:ghost` as an orphan.

## Out of scope for this PR (do not block on these)

- **Confirm delete** of `raw/` — correctly deferred; skip/keep/restore is a valid v1 review loop.
- **Degree ≤ 1** and TTL — #266 listed those as open questions; degree-0 only is the right first slice.
- **MCP `search` / `get_node`** still return parked nodes. Wiki/lint go quiet; agents do not. Fine for v1 if `docs/architecture/serve-mcp.md` (or AGENTS.md) says quarantine is wiki+lint only. Filtering ScopedGraph/FTS can be a follow-up.
- Slugs longer-term: `get_node` itself should return `None` for phantoms rather than KeyError. Not required if lint stops calling it that way.

**Comment by Aohkne** [comment]
> The direction matches #266: park degree-0 nodes with a props flag (not a schema bump, not a mega-merge), persist it across `compile` the same way `merged_ids` is persisted, hide them from wiki while they stay degree-0, and leave `raw/` / confirm-delete for a later design. Tests cover detect/review, lint/heal exclusion, wiki reappearance after an edge appears, and a real `compile` round-trip. Docs + generated CLI reference are in the same PR. Merge onto current `main` is clean; required CI is green.
> 
> One merge-blocking crash, plus a couple of follow-ups that should land with the fix.
> 
> ## Must fix
> `lint()` now calls `store.get_node(nid)` while iterating `store.node_ids()`. `node_ids()` includes NetworkX phantom endpoints created by dangling edges; those nodes have empty node data, so `get_node` raises `KeyError: 'node'`. I reproduced this on the PR tip:
> 
> ```
> GraphStore([svc:a], [edge a → svc:ghost])
> lint(store)  →  KeyError: 'node'
> ```
> 
> That is a regression: `agent lint` / `quarantine detect` (which reuses `lint().orphans`) now crash on the exact broken graphs lint/heal exist to diagnose. Compile output usually has no dangling edges (resolve drops them), but `self_heal` and `lint --auto-fix` exist because they still show up from journals / partial writes.
> 
> Fix: iterate `store.all_nodes()` (already filters phantoms; `self_heal` uses the same set) instead of `node_ids()` + `get_node`:
> 
> ```python
> for n in store.all_nodes():
>     if is_quarantined(n):
>         continue
>     if not store.out_edges(n.id) and not store.in_edges(n.id):
>         report.orphans.append(n.id)
> ```
> 
> Add a regression test: `GraphStore([svc:a], [edge a→ghost])` — `lint` must return, not raise, and must not treat `svc:ghost` as an orphan.
> 
> ## Out of scope for this PR (do not block on these)
> * **Confirm delete** of `raw/` — correctly deferred; skip/keep/restore is a valid v1 review loop.
> * **Degree ≤ 1** and TTL — #266 listed those as open questions; degree-0 only is the right first slice.
> * **MCP `search` / `get_node`** still return parked nodes. Wiki/lint go quiet; agents do not. Fine for v1 if `docs/architecture/serve-mcp.md` (or AGENTS.md) says quarantine is wiki+lint only. Filtering ScopedGraph/FTS can be a follow-up.
> * Slugs longer-term: `get_node` itself should return `None` for phantoms rather than KeyError. Not required if lint stops calling it that way.

Fixed in [8047cf2](https://github.com/manhhailua/lorekeep/commit/8047cf2e48445b5daa57aacdbde284425f778f2f) - lint() now iterates store.all_nodes() instead of node_ids() + get_node(), so dangling-edge phantoms no longer crash it. Added the regression test (test_dangling_edge_does_not_crash) plus two more for quarantine detect --apply / quarantine review coverage. Ready for another look.

File: src/lorekeep/agent.py (hunk 1 above)
Comment (Aohkne) [reply src/lorekeep/agent.py]: Fixed in 8047cf2 — now iterates store.all_nodes() and skips is_quarantined(n) as suggested, plus added the svc:a → svc:ghost regression test.
